### PR TITLE
Support CUDA 11.2 and NCCL 2.8

### PR DESCRIPTION
**Pull request by takagi**
This PR includes changes to support CUDA 11.2. Note that we have to merge #633 first as CUDA 11.2 does not support cuDNN 8.0.

### CODE REVIEW

**Comment by kmaehashi** [comment]
#633 merged! 🚀

**Comment by takagi** [comment]
pfnCI, test this please.

File: docker.py [hunk 1]
```diff
@@ -87,8 +90,9 @@
                 'cudnn76-cuda100'],
     'cuda101': ['cudnn75-cuda101', 'cudnn76-cuda101'],
     'cuda102': ['cudnn76-cuda102'],
-    'cuda110': ['cudnn80-cuda110'],
-    'cuda111': ['cudnn80-cuda111'],
+    'cuda110': ['cudnn80-cuda110', 'cudnn81-cuda110'],
+    'cuda111': ['cudnn80-cuda111', 'cudnn81-cuda111'],
+    'cuda112': ['cudnn80-cuda112', 'cudnn81-cuda112'],
```
Comment (kmaehashi) [inline docker.py:95]:
```suggestion
    'cuda112': ['cudnn81-cuda112'],
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docker.py [hunk 2]
```diff
@@ -647,158 +669,189 @@ def get_scipy_choices():
...
     libnccl2='libnccl2_2.7.8-1+cuda11.1_amd64',
     libnccl_dev='libnccl-dev_2.7.8-1+cuda11.1_amd64',
     include_dir='/usr/include',
     lib_dir='/usr/lib/x86_64-linux-gnu',
 )
 
+codes['nccl2.8-cuda112'] = nccl_base.format(
+    nccl_os_ver='1804',
+    libnccl2='libnccl2_2.8.3-1+cuda11.2_amd64.deb',
+    libnccl_dev='libnccl-dev_2.8.3-1+cuda11.2_amd64.deb',
```
Comment (kmaehashi) [inline docker.py:842]:
```suggestion
    libnccl2='libnccl2_2.8.4-1+cuda11.2_amd64',
    libnccl_dev='libnccl-dev_2.8.4-1+cuda11.2_amd64',
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docker.py [hunk 3]
```diff
@@ -647,158 +669,189 @@ def get_scipy_choices():
...
     libnccl2='libnccl2_2.7.8-1+cuda11.1_amd64',
     libnccl_dev='libnccl-dev_2.7.8-1+cuda11.1_amd64',
     include_dir='/usr/include',
     lib_dir='/usr/lib/x86_64-linux-gnu',
 )
 
+codes['nccl2.8-cuda112'] = nccl_base.format(
+    nccl_os_ver='1804',
+    libnccl2='libnccl2_2.8.4-1+cuda11.2_amd64',
+    libnccl_dev='libnccl-dev_2.8.4-1+cuda11.2_amd64',
```
Comment (kmaehashi) [inline docker.py:842]:
```suggestion
    libnccl2='libnccl2_2.8.3-1+cuda11.2_amd64',
    libnccl_dev='libnccl-dev_2.8.3-1+cuda11.2_amd64',
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
CI is unrelated

**Pull request merged**